### ISSUE TITLE
Using multiple datasets with batch size bigger than 1

### ISSUE THREAD

**Issue by chanil1218** [kohya-ss/musubi-tuner#824]
Using batch size bigger than 1 instead of using gradient accumulation, 
single batch consists of data from the single dataset, instead of mixing samples from datasets. 
I saw the line "do no shuffle here to avoid multiple datasets have different order". 

But I think this behavior degrades batch gradient stability like fluctuation or oscillation. 

So I forced to use gradient accumulation instead of increasing the batch size.
But I thought this would have potential issue of batch statistics would not be utilized optimally and the training speed.
